readme.txt
==== Nona ====
=== Readme.Txt ===
Last revised February 2016

== Table of Contents ==
* Recent Changelog
* To-Do
* Notes
* Questions & Answers
* Review Tickets

== Recent Changelog ==
== Version 1.9.3 ==
* Released February 2016
* Update copyright year
* Update required version to WordPress 4.1 (no sanity check)
* Added `id` to sidebar widget definitions
* Added sanity check before registering widgets
* Added `screen-reader-text` required CSS definitions
* Remove extraneous/minor comments to clean up code

== Version 1.9.2 ==
Changelog: December 27, 2014
= Code =
* Additional minor code refactoring to better meet WordPress Coding Standards
* Added `nona_content_width` function to be used in `*_head` hooks
* Added sanity check for WordPress 4.1 `add_theme_support( 'title-tag' )`
* Removed call to unused `$wp_version` global in `nona_setup`

= CSS =
* Re-implemented `.hidden` class to use `display: none;` property

= Miscellaneous =
* Added BNS Login compatibility to display dashicons instead of text links
* Cleaned up theme description

/** ------------------------------------------------------------------------- */
== Version 1.9.1 ==
Changelog: August 4, 2014
= Code =
* Minor code refactoring to better meet WordPress Coding Standards

= CSS =
* Added `post-title` class and styles to better handle long titles
* Improved aesthetics of Post Password form display
* Improved aesthetics of Search form display
* Made sure Twitter embeds are not fixed width
* Minor adjustment to child-menu item alignment (now left instead of centered)
* Updated `gallery-caption` class to better handle high quantity column displays
* Updated default Calendar widget styles

= Miscellaneous =
* Documentation clean up and updates
* Update copyright to 2014
* Update Tested Up To version to WordPress 4.0
* Update `screenshot.png` to 880x660 size

/** ------------------------------------------------------------------------- */
== Version 1.9 ==
Changelog: December 28, 2013
= Code =
* Code reformatted to better meet WordPress Coding Standards (see https://gist.github.com/Cais/8023722)
* Remove `nona_login` function - see BNS Login http://wordpress.org/extend/plugins/bns-login
* Removed the unused variable `$sep_location`

/** ------------------------------------------------------------------------- */
For older changelog entries see `changelog.txt`

== To-Do ==
NB: Open To-Do List is now found in 'index.php'

== Notes ==
* Future changelog entries will reflect commits to the GitHub repository: https://github.com/Cais/nona
* Some very light background colors will make the links hard to read when hovering over them. See the CSS element `a:hover { color: #e3e3e3; }` in style.css if you want to change this color.

== Questions & Answers ==
* TBA

== Review Tickets ==
* http://themes.trac.wordpress.org/ticket/2582
* http://themes.trac.wordpress.org/ticket/5740 - version 1.4
* http://themes.trac.wordpress.org/ticket/7530 - version 1.5
* http://themes.trac.wordpress.org/ticket/8575 - version 1.6
* http://themes.trac.wordpress.org/ticket/10307 - version 1.7
* http://themes.trac.wordpress.org/ticket/11704 - version 1.8 - March 2013
* http://themes.trac.wordpress.org/ticket/13448 - version 1.8.1 - July 2013
* http://themes.trac.wordpress.org/ticket/15897 - version 1.9 - December 28, 2013
* https://themes.trac.wordpress.org/ticket/20075 - version 1.9.1 - August 4, 2014
* https://themes.trac.wordpress.org/ticket/22342 - version 1.9.2 - December 27, 2014

changelog.txt
=== NoNa ===
== Version 1.9.3 ==
* Released February 2016
* Update copyright year
* Update required version to WordPress 4.1 (no sanity check)
* Added `id` to sidebar widget definitions
* Added sanity check before registering widgets
* Added `screen-reader-text` required CSS definitions
* Remove extraneous/minor comments to clean up code

/** ------------------------------------------------------------------------- */
== Version 1.9.2 ==
Changelog: December 27, 2014
= Code =
* Additional minor code refactoring to better meet WordPress Coding Standards
* Added `nona_content_width` function to be used in `*_head` hooks
* Added sanity check for WordPress 4.1 `add_theme_support( 'title-tag' )`
* Removed call to unused `$wp_version` global in `nona_setup`

= CSS =
* Re-implemented `.hidden` class to use `display: none;` property

= Miscellaneous =
* Added BNS Login compatibility to display dashicons instead of text links
* Cleaned up theme description

/** ------------------------------------------------------------------------- */
== Version 1.9.1 ==
Changelog: August 4, 2014
= Code =
* Minor code refactoring to better meet WordPress Coding Standards

= CSS =
* Added `post-title` class and styles to better handle long titles
* Improved aesthetics of Post Password form display
* Improved aesthetics of Search form display
* Made sure Twitter embeds are not fixed width
* Minor adjustment to child-menu item alignment (now left instead of centered)
* Updated `gallery-caption` class to better handle high quantity column displays
* Updated default Calendar widget styles

= Miscellaneous =
* Documentation clean up and updates
* Update copyright to 2014
* Update Tested Up To version to WordPress 4.0
* Update `screenshot.png` to 880x660 size

/** ------------------------------------------------------------------------- */
== Version 1.9 ==
Changelog: December 28, 2013
= Code =
* Code reformatted to better meet WordPress Coding Standards (see https://gist.github.com/Cais/8023722)
* Remove `nona_login` function - see BNS Login http://wordpress.org/extend/plugins/bns-login
* Removed the unused variable `$sep_location`

/** ------------------------------------------------------------------------- */
== Version 1.8.1 ==
Changelog: July 18, 2013
= Code =
* Added constant 'NONA_HOME_URL'
* Hide middle footer widget area frame if it is empty
* Refactored Featured Image display code into function `nona_show_featured_image( $size )`
* Used 'NONA_HOME_URL' where appropriate

= CSS =
* Added 'hidden' class
* Adjusted `img` and 'caption' related properties

/** ------------------------------------------------------------------------- */
== Version 1.8 ==
Changelog: March 15, 2013
= Code =
* Added code block termination comments
* Changed conditional to use `is_multisite` for Multisite testing
* Move Comment Reply script enqueue to 'functions.php'
* Refactored code formatting
* Refactored no post title code
* Refactored to be more i18n compatible
* Refactored widget area definitions into `nona_widgets` function and attached to `widgets_init` action hook
* Remove 'searchform.php' template in favor of using WordPress core version
* Updated DOCTYPE and other related header elements

= CSS =
* Minor adjustments to handle default widgets (based on the "Monster" widget)
* Minor adjustment to handle default 'searchform' output

= Miscellaneous =
* Deprecate `nona_login`, recommend using plugin BNS Login instead.
* Update "WordPress Required Version" to 3.4

/** ------------------------------------------------------------------------- */
== Version 1.7 ==
Changelog: December 10, 2012
= Code =
* Fixed spacing issue in meta details if post has no tags
* Refactored sidebar definitions to include names and descriptions for all areas
* Refactored `wp_nav_menu` support code

= Miscellaneous =
* PHPDocs style documentation updates and additions
* Minor change to output text in 'NoNa Theme Version'

/** ------------------------------------------------------------------------- */
== Version 1.6 ==
Changelog: July 10, 2012
= Code =
* Added `nona_wp_title` to be used with `wp_title` filter
* Removed deprecated function call to `add_custom_background`
* Removed deprecated function call to `get_theme_data`
* Updated post to post navigation in 'single.php'

= CSS =
* Miscellaneous minor style changes
* Styles for default widgets updated

= Internet Browsers Reviewed =
* Apple Safari v5.1.7
* Firefox v13.0.1
* Google Chrome v20.0
* Internet Explorer v9.0
* Opera v12.0

/** ------------------------------------------------------------------------- */
== Version 1.5 ==
Changelog: April 24, 2012
= Code =
* Added $args to `nona_login` function - adopted from BNS-Login plugin
* Added conditional check to only display website (and email) if exists, as well as only display biography if it exists
* Addressed deprecated function call to `add_custom_background`
* Addressed deprecated function calls to `get_theme_data`
* Make post edit link preceding character consistent across all theme templates

= CSS =
* Corrected various class names ... changed from 'bns-' to 'nona-'
* Form aesthetics improved

/** ------------------------------------------------------------------------- */
== Version 1.4 ==
Changelog: November 6, 2011
= Code =
* Added textdomain as needed to string functions
* Added previous and next post navigation in the `single.php` template
* Added `wp_link_pages` after every instance of `the_content`
* Added `description` parameter to `register_sidebar` definitions
* Removed `show_avatar` function - not used
* Removed `bns_menu` function - not used
* Removed `nona_get_page_number` function - no longer used
* Removed display of 'Featured Image' in `single.php` for aesthetic reasons
* Removed default 'Home' link in menus when not on home page; leave as end-user choice via custom menu
* Replaced `the_author_posts_link` (deprecated) with `get_author_posts_url`
* Updated 'BNS Dynamic Copyright' and renamed to 'NoNa Dynamic Copyright'

author.php
<?php
/**
 * Author Template
 *
 * @package     NoNa
 * @since       1.0
 *
 * @link        http://buynowshop.com/themes/nona/
 * @link        https://github.com/Cais/nona/
 * @link        https://wordpress.org/themes/nona
 *
 * @author      Edward Caissie <edward.caissie@gmail.com>
 * @copyright   Copyright (c) 2009-2016, Edward Caissie
 *
 * Last revised April 23, 2012
 * @version     1.5
 * Added conditional check to only display website (and email) if exists, as well as only display biography if it exists
 *
 * @version     1.8
 * @date        March 14, 2013
 * Added code block termination comments
 * Refactored code formatting
 * Refactored to be more i18n compatible
 */

get_header();

/** Set the $curauth variable */
$curauth = ( get_query_var( 'author_name ' ) )
	? get_user_by( 'id', get_query_var( 'author_name' ) )
	: get_userdata( get_query_var( 'author' ) ); ?>

	<div id="main-blog">

		<div id="before-content"></div>

		<div id="content">

			<div id="author" class="<?php if ( user_can( $curauth, 'manage_options' ) ) {
				echo 'administrator';
			} ?>">

				<h2>
					<?php printf( __( 'About %1$s', 'nona' ), $curauth->display_name ); ?>
				</h2>

				<ul>
					<?php if ( ! empty( $curauth->user_url ) ) { ?>
						<li>
							<?php
							printf(
								__( 'Website %1$s or %2$s', 'nona' ),
								sprintf( '<a href="%1$s">%1$s</a>', $curauth->user_url ),
								sprintf( '<a href="mailto:%1$s">email</a>', $curauth->user_email )
							); ?>
						</li>
						<?php
					}

					if ( ! empty( $curauth->user_description ) ) {
						?>
						<li>
							<?php printf( __( 'Biography: %1$s', 'nona' ), $curauth->user_description ); ?>
						</li>
					<?php } ?>
				</ul>

			</div>
			<!-- #author -->

			<h2>
				<?php printf( __( 'Posts by %1$s:', 'nona' ), $curauth->display_name ); ?>
			</h2>

			<!-- start the Loop -->
			<?php
			if ( have_posts() ) {
				$count = 0;
				while ( have_posts() ) {
					the_post();
					$count ++; ?>

					<div <?php post_class(); ?> id="post-<?php the_ID(); ?>">

						<h2>
							<a href="<?php the_permalink() ?>" rel="bookmark" title="<?php _e( 'Permanent Link to', 'nona' ); ?> <?php the_title_attribute(); ?>"><?php the_title(); ?></a>
						</h2>

						<div class="post-details">

							<?php
							printf(
								__( 'on %1$s ', 'nona' ),
								get_the_time( get_option( 'date_format' ) )
							);

							if ( ! post_password_required() ) {
								echo ' ';
								comments_popup_link( __( 'with No Comments', 'nona' ), __( 'with 1 Comment', 'nona' ), __( 'with % Comments', 'nona' ), '', __( 'with Comments Closed', 'nona' ) );
							}

							edit_post_link( __( 'Edit', 'nona' ), __( ' | ', 'nona' ), __( '', 'nona' ) );

							printf(
								__( '<div class="nona-categories-list">in %1$s</div>', 'nona' ),
								get_the_category_list( ', ' )
							);
							the_tags( __( 'as ', 'nona' ), ', ', '' ); ?>

						</div>
						<!-- .post-details -->

						<?php
						nona_show_featured_image( 'full' );

						if ( $count == 1 ) {
							the_content( __( 'Read more... ', 'nona' ) ); ?>
							<div class="clear"></div><!-- For inserted media at the end of the post -->
							<?php
							wp_link_pages(
								array(
									'before'         => '<p><strong>' . __( 'Pages: ', 'nona' ) . '</strong>',
									'after'          => '</p>',
									'next_or_number' => 'number'
								)
							);
						} else {
							the_excerpt(); ?>
							<div class="clear"></div><!-- For inserted media at the end of the post -->
						<?php } ?>

					</div><!-- .post #post-ID -->

				<?php } ?>

				<div id="nav-global" class="navigation">
					<div class="left">
						<?php next_posts_link( __( '&laquo; Previous entries ', 'nona' ) ); ?>
					</div>
					<div class="right">
						<?php previous_posts_link( __( ' Next entries &raquo;', 'nona' ) ); ?>
					</div>
				</div><!-- .navigation -->

				<div class="clear"></div>

			<?php } else { ?>

				<h2>
					<?php
					printf(
						__( 'Search Results for: %s', 'nona' ),
						'<span>' . esc_html( get_search_query() ) . '</span>'
					); ?>
				</h2>

				<p><?php _e( 'Sorry, but you are looking for something that is not here.', 'nona' ); ?></p>

				<?php
				get_search_form();

			} ?>
			<!-- end the Loop -->

		</div>
		<!-- #content -->

		<div id="after-content"></div>

	</div><!-- #main-blog -->

<?php
get_sidebar();
get_footer();